kpi/tests/test_assets.py

```diff
@@ -779,3 +781,85 @@ def test_anonymous_as_baseline_for_authenticated(self):
         for user_obj in AnonymousUser(), self.anotheruser:
             self.assertTrue(user_obj.has_perm(
                 PERM_VIEW_ASSET, self.asset))
+
+
+class TestAssetNameSettingHandling(AssetsTestCase):
+    """
+    Tests for the 'name' setting in the asset content
+    """
+    def test_asset_name_matches_instance_root(self):
+        """
+        Test if 'name' setting is provided, it should match with the root node.
+        """
+        content = {
+            'survey': [
+                {
+                    'type': 'select_one',
+                    'label': 'q1',
+                    'select_from_list_name': 'iu0sl99',
+                },
+            ],
+            'choices': [
+                {'name': 'a11', 'label': ['a11'], 'list_name': 'iu0sl99'},
+                {'name': 'a3', 'label': ['a3'], 'list_name': 'iu0sl99'},
+            ],
```

Comment:
Please change the `type` to something simple like `text`, and then you can eliminate `select_from_list_name` as well as `choices`. It'll just make the test easier to read since we're not testing any features related to `select` questions